packages/next/src/client/components/client-page.tsx

```diff
@@ -1,30 +1,76 @@
 'use client'
 
+import type { ParsedUrlQuery } from 'querystring'
+import { InvariantError } from '../../shared/lib/invariant-error'
+
+import type { Params } from '../../server/request/params'
+
+/**
+ * When the Page is a client component we send the params and searchParams to this client wrapper
+ * where they are turned into dynamically tracked values before being passed to the actual Page component.
+ *
+ * additionally we may send promises representing the params and searchParams. We don't ever use these passed
+ * values but it can be necessary for the sender to send a Promise that doesn't resolve in certain situations.
+ * It is up to the caller to decide if the promises are needed.
+ */
 export function ClientPageRoot({
   Component,
-  props,
+  searchParams,
+  params,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  promises,
```

Comment:
Maybe instead we should create new component that does use this promise instead? It's also not needed to define it here so long as it's defined in the types.

Reply:
I did originally but it makes some of the psuedo-stacks for hydration diffs worse so I kept it in this one component